lib/src/core/transaction_signer.dart

```
@@ -1,26 +1,26 @@
-// @dart=2.9
 part of 'package:web3dart/web3dart.dart';
 
 class _SigningInput {
+  _SigningInput(
+      {required this.transaction, required this.credentials, this.chainId});
+
   final Transaction transaction;
   final Credentials credentials;
-  final int chainId;
-
-  _SigningInput({this.transaction, this.credentials, this.chainId});
+  final int? chainId;
 }
 
 Future<_SigningInput> _fillMissingData({
-  @required Credentials credentials,
-  @required Transaction transaction,
-  int chainId,
+  required Credentials credentials,
+  required Transaction transaction,
+  int? chainId,
   bool loadChainIdFromNetwork = false,
-  Web3Client client,
+  Web3Client? client,
 }) async {
-  assert(credentials != null);
-  assert(transaction != null);
-  assert(loadChainIdFromNetwork != null);
-  assert(!loadChainIdFromNetwork || chainId != null,
-      "You can't specify loadChainIdFromNetwork and specify a custom chain id!");
+  // assert(credentials != null);
+  // assert(transaction != null);
+  // assert(loadChainIdFromNetwork != null);
+  // assert(!loadChainIdFromNetwork || chainId != null,
+  //     "You can't specify loadChainIdFromNetwork and specify a custom chain id!");
```

This one should still be kept though.

---

Asserts only run in development?

---

Good point, it should throw an argument error.